conform-pr/main.go

```diff
@@ -28,52 +28,62 @@ func main() {
 		action.Fatalf("main: %s", err)
 	}
 
-	errors := runChecks(action, client)
-
-	if len(errors) == 0 {
-		return
+	summaries := runChecks(action, client)
+	action.AddStepSummary("| Check  | Status |")
+	action.AddStepSummary("|--------|--------|")
+
+	for _, summary := range summaries {
+		statusSign := ":heavy_multiplication_x:"
+		if summary.Ok {
+			statusSign = ":heavy_check_mark:"
+		}
+		if summary.Details != nil {
+			action.AddStepSummary(fmt.Sprintf("|%s | %s %s|", summary.Name, statusSign, summary.Details))
+		} else {
+			action.AddStepSummary(fmt.Sprintf("|%s | %s |", summary.Name, statusSign))
+		}
 	}
 
-	var serrors []string
-	for _, err := range errors {
-		serrors = append(serrors, fmt.Sprintf("%s", err))
+	for _, v := range summaries {
+		if !v.Ok {
+			action.Fatalf("The PR does not conform to the rules")
+		}
 	}
+}
 
-	action.Fatalf("The PR does not conform to the rules:\n - %s.", strings.Join(serrors, ";\n - "))
+type Summary struct {
+	Name    string
+	Ok      bool
+	Details error
 }
 
 // runChecks runs all the checks included into the PR conformance rules.
-// It returns the list of errors that occurred during the checks.
-func runChecks(action *githubactions.Action, client graphql.Querier) []error {
-	var errors []error
-
-	pr, err := getPR(action, client)
-	if err != nil {
-		return []error{fmt.Errorf("runChecks: %w", err)}
+// It returns the list of check summary for the checks.
+func runChecks(action *githubactions.Action, client graphql.Querier) []Summary {
+	pr, summaries := getPR(action, client)
+	if len(summaries) > 0 {
+		return summaries
 	}
 
 	// PRs from dependabot are perfect
 	if pr.author == "dependabot[bot]" {
 		return nil
 	}
 
-	if err := pr.checkTitle(); err != nil {
-		errors = append(errors, err)
-	}
+	summaries = pr.checkTitle()
 
-	if err := pr.checkBody(action); err != nil {
-		errors = append(errors, err)
-	}
+	summaries = append(summaries, pr.checkBody(action)...)
 
-	return errors
+	return summaries
 }
 
-// getPR returns PR's information.
-// If an error occurs, it returns nil and the error.
-func getPR(action *githubactions.Action, client graphql.Querier) (*pullRequest, error) {
+// getPR returns PR's information and returns
+// * pull request details if no errors occured
+// * a summary list whether and which check passed successfully or not.
+func getPR(action *githubactions.Action, client graphql.Querier) (*pullRequest, []Summary) {
```

Comment:
I would expect that function to return PR info (as it did before) and let's the caller care about summaries, etc